app/dashboard/views.py

```diff
@@ -3442,7 +3443,145 @@ def hackathon_onboard(request, hackathon=''):
+            Q(name__icontains=q) |
+            Q(summary__icontains=q) |
+            Q(profiles__handle__icontains=q)
+        )
+    if filters == 'winners':
+        projects = projects.filter(
+            Q(badge__isnull=False)
+        )
+
+    projects_paginator = Paginator(projects, 9)
+
+    try:
+        projects_paginated = projects_paginator.page(page)
+    except PageNotAnInteger:
+        projects_paginated = projects_paginator.page(1)
+    except EmptyPage:
+        projects_paginated = projects_paginator.page(projects_paginator.num_pages)
+
+    params = {
+        'active': 'hackathon_onboard',
+        'title': 'Hackathon Projects',
+        'hackathon': hackathon_event,
+        'is_registered': is_registered,
+        'projects': projects_paginated,
+        'order_by': order_by,
+        'filters': filters,
+        'query': q.split
+    }
+
+    return TemplateResponse(request, 'dashboard/hackathon/projects.html', params)
+
+
+@csrf_exempt
+def hackathon_get_project(request, bounty_id, project_id=None):
+    # TODO FIX cache problem
+
+    profile = request.user.profile if request.user.is_authenticated and hasattr(request.user, 'profile') else None
+    try:
+        bounty = Bounty.objects.current().get(id=bounty_id)
+        projects = HackathonProject.objects.filter(bounty__standard_bounties_id=bounty.standard_bounties_id, profiles__id=profile.id).nocache()
+        print(projects)
+    except HackathonProject.DoesNotExist:
+        pass
+
+    if project_id:
+        project_selected = projects.filter(id=project_id).first()
+    else:
+        project_selected = None
+
+    params = {
+        'bounty_id': bounty_id,
+        'bounty': bounty,
+        'projects': projects,
+        'project_selected': project_selected
+    }
+    return TemplateResponse(request, 'dashboard/hackathon/project_new.html', params)
+
+
+@csrf_exempt
+@require_POST
+def hackathon_save_project(request):
+
+    project_id = request.POST.get('project_id')
+    bounty_id = request.POST.get('bounty_id')
+    profiles = request.POST.getlist('profiles[]')
+    logo = request.FILES.get('logo')
+    profile = request.user.profile if request.user.is_authenticated and hasattr(request.user, 'profile') else None
+
+    error_response = invalid_file_response(logo, supported=['image/png', 'image/jpeg', 'image/jpg'])
+    # 400 is ok because file upload is optional here
+    if error_response and error_response['status'] != 400:
+        return JsonResponse(error_response)
+
+    if profile is None:
+        return JsonResponse({
+            'success': False,
+            'msg': '',
+        })
+
+    bounty_obj = Bounty.objects.current().get(pk=bounty_id)
+
+    kwargs = {
+        'name': request.POST.get('name'),
+        'hackathon': bounty_obj.event,
+        'logo': request.FILES.get('logo'),
+        'bounty': bounty_obj,
+        'summary': request.POST.get('summary'),
+        'work_url': request.POST.get('work_url')
```

Comment:
Is there a danger someone may put something malicious in this args? We probably need to sanitize them